pkg/controller/provider/container/vsphere/model.go

```diff
@@ -398,6 +400,19 @@ func (v *NetworkAdapter) Apply(u types.ObjectUpdate) {
 				if s, cast := p.Val.(types.OpaqueNetworkSummary); cast {
 					v.model.Key = s.OpaqueNetworkId
 				}
+			case fDVSwitchVlan:
+				if portSettings, cast := p.Val.(types.VMwareDVSPortSetting); cast {
+					if vlanIdSpec, cast := portSettings.Vlan.(*types.VmwareDistributedVirtualSwitchVlanIdSpec); cast && int(vlanIdSpec.VlanId) > 0 {
+						v.model.VlanId = strconv.Itoa(int(vlanIdSpec.VlanId))
+					}
+					if vlanIdSpec, cast := portSettings.Vlan.(*types.VmwareDistributedVirtualSwitchTrunkVlanSpec); cast {
+						refList := []string{}
+						for _, val := range vlanIdSpec.VlanId {
+							refList = append(refList, fmt.Sprintf("%d-%d", val.Start, val.End))
+						}
+						v.model.VlanId = strings.Join(refList, ",")
+					}
+				}
```

Comment:
Could you please use the switchcase for the `portSettings.Vlan` types?
In case we missed some types it will be easier to add in future 

```
					switch portSettings.Vlan.(type) {
					case *types.VmwareDistributedVirtualSwitchVlanIdSpec:
						if cast && int(vlanIdSpec.VlanId) > 0 {
							v.model.VlanId = strconv.Itoa(int(vlanIdSpec.VlanId))
						}
					case *types.VmwareDistributedVirtualSwitchTrunkVlanSpec:
						refList := []string{}
						for _, val := range vlanIdSpec.VlanId {
							refList = append(refList, fmt.Sprintf("%d-%d", val.Start, val.End))
						}
						v.model.VlanId = strings.Join(refList, ",")
					}
```